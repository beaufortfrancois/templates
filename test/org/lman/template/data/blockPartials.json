{
  "apple": "yummy apple",
  "banana": "yummy banana",
  "veggies": {
    "carrot": "yummy carrot"
  }
}
