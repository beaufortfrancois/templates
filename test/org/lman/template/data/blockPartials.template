{{+content:partials.wrapper}}
This content is passed to |wrapper| mapped to content.
{{/partials.wrapper}}

{{+content:partials.wrapper fruit:apple}}
Likewise, but this time there's a |fruit| argument.
It can't be accessed from here {{?fruit}}OOPS{{/}} but the outer scope can still be
accessed ({{apple}}).
{{/partials.wrapper}}

{{#veggies}}
{{+content:partials.wrapper}}
{{- TODO: fix this. The partial needs to pass its local context on. -}}
Here, the context from |veggies| should give us access to {{- carrot -}}.
{{/partials.wrapper}}
{{/veggies}}

{{+content:partials.wrapper}}
And look! The wrapper is passing us back a {{veg}}.
{{/partials.wrapper}}
