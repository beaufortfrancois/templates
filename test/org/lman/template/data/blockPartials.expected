Wrapping:

This content is passed to |wrapper| mapped to content.

Wrapping:

Likewise, but this time there's a |fruit| argument.
It can't be accessed from here  but the outer scope can still be
accessed (yummy apple).
With yummy apple!

Wrapping:

Here, the context from |veggies| should give us access to yummy carrot.

Wrapping:

And look! The wrapper is passing us back a yummy carrot.
